src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -312,28 +312,30 @@ protected async Task<Solution> PostProcessChangesAsync(Solution changedSolution,
         /// <returns>A document with the post processing changes applied.</returns>
         protected virtual async Task<Document> PostProcessChangesAsync(Document document, CancellationToken cancellationToken)
         {
-            var options = await CodeCleanupOptions.FromDocumentAsync(document, fallbackOptions: null, cancellationToken).ConfigureAwait(false);
-            return await CleanupDocumentAsync(document, options, cancellationToken).ConfigureAwait(false);
+            if (document.SupportsSyntaxTree)
+            {
+                var options = await CodeCleanupOptions.FromDocumentAsync(document, fallbackOptions: null, cancellationToken).ConfigureAwait(false);
+                return await CleanupDocumentAsync(document, options, cancellationToken).ConfigureAwait(false);
+            }
+
+            return document;
         }
 
         internal static async Task<Document> CleanupDocumentAsync(
             Document document, CodeCleanupOptions options, CancellationToken cancellationToken)
         {
-            if (document.SupportsSyntaxTree)
```

Comment:
Assert `document.SupportsSyntaxTree`?

Reply:
The calls below throw if services are not available, so an additional assert is not needed imo.

Reply:
Sounds good